### ISSUE TITLE
[Bug]: mixcoord panic: runtime error: index out of range [0] with length 0

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#29699]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-io-master-dc8b5c1-20240104
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus with config
```
  config:
    dataCoord:
      enableActiveStandby: true
      segment:
        enableLevelZero: true
    indexCoord:
      enableActiveStandby: true
    log:
      level: debug
    queryCoord:
      enableActiveStandby: true
    rootCoord:
      enableActiveStandby: true
```
2. create collection -> index -> insert 6m data into 3 partitions -> flush -> index -> load
3. concurrent search + insert + delete + flush
4. mixcoord panic
[mc_d557l.log](https://github.com/milvus-io/milvus/files/13831486/mc_d557l.log)

```
[2024/01/04 14:54:25.782 +00:00] [INFO] [observers/collection_observer.go:237] ["load status updated"] [collectionID=446792473996165537] [partitionID=446792473999366045] [partitionLoadPercentage=100] [collectionLoadPercentage=66]
[2024/01/04 14:54:25.782 +00:00] [INFO] [observers/collection_observer.go:191] ["partition targets"] [collectionID=446792473996165537] [partitionID=446792474001966548] [segmentTargetNum=4] [channelTargetNum=2] [totalTargetNum=6] [replicaNum=1]
[2024/01/04 14:54:25.782 +00:00] [INFO] [observers/collection_observer.go:214] ["partition load progress"] [collectionID=446792473996165537] [partitionID=446792474001966548] [subChannelCount=2] [loadSegmentCount=4]
[2024/01/04 14:54:25.784 +00:00] [INFO] [observers/collection_observer.go:237] ["load status updated"] [collectionID=446792473996165537] [partitionID=446792474001966548] [partitionLoadPercentage=100] [collectionLoadPercentage=100]
panic: runtime error: index out of range [0] with length 0

goroutine 698 [running]:
panic({0x4db0580, 0xc002f21a58})
    /usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc0036153b8 sp=0xc0036152f8 pc=0x1a8bc9b
runtime.goPanicIndex(0x0, 0x0)
    /usr/local/go/src/runtime/panic.go:113 +0x7f fp=0xc0036153f8 sp=0xc0036153b8 pc=0x1a899df
github.com/milvus-io/milvus/internal/querycoordv2/balance.heapQueue.Swap(...)
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/priority_queue.go:51
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*heapQueue).Swap(0x2e94f3?, 0x2e94f3?, 0x263b58?)
    <autogenerated>:1 +0xc5 fp=0xc003615418 sp=0xc0036153f8 pc=0x4448125
container/heap.Pop({0x564a560, 0xc002e49a40})
    /usr/local/go/src/container/heap/heap.go:61 +0x45 fp=0xc003615450 sp=0xc003615418 pc=0x23f2ea5
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*priorityQueue).pop(...)
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/priority_queue.go:84
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*ScoreBasedBalancer).AssignSegment(0x562ac20?, 0xc00311c000?, {0xc00311d000?, 0x1bc, 0x200}, {0xc002d8f800?, 0xc0022f2200?, 0x48f70a0?})
```

### Steps To Reproduce

```markdown
argo: [level-zero-stable-7-3164711395](https://argo-workflows.zilliz.cc/workflows/qa/level-zero-stable-7?tab=workflow&nodeId=level-zero-stable-7-3164711395&nodePanelView=inputs-outputs)
```

### Milvus Log

pods:
```
level-zero-stable-dml-5-milvus-datanode-5dbd6b9ccf-5pvwq          1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-datanode-5dbd6b9ccf-82djh          1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-indexnode-7b78db5bf4-jz4fq         1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-indexnode-7b78db5bf4-v6bvz         1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-mixcoord-cc7d9699d-d557l           0/1     CrashLoopBackOff   10 (96s ago)      41m
level-zero-stable-dml-5-milvus-proxy-6bcd9b4996-tzz8n             1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-querynode-85887d7d45-4pbp9         1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-querynode-85887d7d45-66bmv         1/1     Running            0                 41m
level-zero-stable-dml-5-milvus-querynode-85887d7d45-qghl2         1/1     Running            0                 41m
```

**Comment by congqixia** [comment]
fix patch has been merged, could you please verify? @ThreadDao 
/assign @ThreadDao

**Comment by ThreadDao** [comment]
Fixed `master-20240108-3d07b668-amd64`
